[Ma Xizi] iP

Ma-Xizi opened the pull request:
## Xizi Chatbot - Enhancements and New Features 🚀 :rabbit:

> “Your mind is for having ideas, not holding them.” – David Allen (source)

**Xizi** is a personal assistant chatbot designed to handle various user commands and manage tasks efficiently. This PR introduces several enhancements to improve functionality and user experience. It is 

- text-based
- easy to learn
- ~~FAST~~SUPER FAST to use!

### New Features:

- [x]  **Task Management**: Add, delete, and mark tasks as done.
- [x]  **Command Handling**: Recognizes and processes various commands such as `mark`, `unmark`, `todo`, `deadline`, and `event`.
- [x]  **Locate Task Fast and Swift**: Find <keyword> to search for all tasks with the given keyword.

### Instructions:
To use the latest version of Xizi, follow these steps:

1. Download the latest release from [here](https://github.com/Ma-Xizi/ip/releases/tag/A-Jar).
2. Install the software by downloading the JAR file.
3. Run the application and start managing your tasks!

### Code Example:

Here’s an example of how to use the `Xizi` chatbot, running the `main` method:

``` ruby
public static void main(String[] args) {
        new Xizi(FILE_PATH).run();
 }

```

### Known Issues:
⚠️ Some commands may not work as expected in certain edge cases.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Ma-Xizi committed:
Level 0. Rename, Greet, Exit

Ma-Xizi committed:
implemented the functionality of level 1

Ma-Xizi committed:
implemented level2

Ma-Xizi committed:
implement level3

Ma-Xizi committed:
implemented Level 4

Ma-Xizi committed:
A-TextUiTesting: Automated Text UI Testing

Ma-Xizi committed:
Level 5. Handle Errors

Ma-Xizi committed:
Level 6. Delete

Ma-Xizi committed:
Improve and take care of Git and Java format requirements

Ma-Xizi committed:
Implement level-7

Ma-Xizi committed:
Update README.md

Ma-Xizi committed:
Update deadline details in README.md

Ma-Xizi committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Update README.md

Ma-Xizi committed:
Add enums for command type

Ma-Xizi committed:
Improve coding syle

Ma-Xizi committed:
Improve java coding style

Ma-Xizi committed:
Modify date storage in task classes

Ma-Xizi committed:
Update command handling and storage about date and time

Ma-Xizi committed:
Correct mistake about skipping the first line

Ma-Xizi committed:
Improve coding style

Ma-Xizi committed:
Improve coding style

Ma-Xizi committed:
Implement function to print deadlines/events occurring on a specific date

Ma-Xizi committed:
Resolve error of falsely reporting no tasks being found

Ma-Xizi committed:
Remove one blank line

Ma-Xizi committed:
Merge branch 'branch-Level-8'

Ma-Xizi committed:
Remove one blank line

Ma-Xizi committed:
Add UI class

Ma-Xizi committed:
Add parser class

Ma-Xizi committed:
Test with data/xizi.txt

Ma-Xizi committed:
Refactor command classes

Ma-Xizi committed:
Put classes into packages

Ma-Xizi committed:
Manage imports and create task package

Ma-Xizi committed:
Merge branch 'add-gradle-support' of https://github.com/Ma-Xizi/ip

* 'add-gradle-support' of https://github.com/Ma-Xizi/ip:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Ma-Xizi committed:
Test with gradle build and make slight adjustments

Ma-Xizi committed:
Refactor the packages

Ma-Xizi committed:
Add tests related to commands

Ma-Xizi committed:
Modify build.gradle with the right names for main class and jar files

Ma-Xizi committed:
Add JavaDoc comments to three classes

Ma-Xizi committed:
Add list command

Ma-Xizi committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

* branch-A-JavaDoc:
  Add JavaDoc comments to three classes

# Conflicts:
#	data/xizi.txt

Ma-Xizi committed:
Update README.md

Ma-Xizi committed:
Add JavaDoc comments for bye command and command interface

Ma-Xizi committed:
Add deadline command java doc

Ma-Xizi committed:
Update some imports logic flow

Ma-Xizi committed:
Update JavaDoc comments and help message

Ma-Xizi committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Update some imports logic flow

Ma-Xizi committed:
Revert "Remove one blank line"

This reverts commit 35baf7f4abcbda42779bd581339624acea439cc0.

Ma-Xizi committed:
Touch up imports

Ma-Xizi committed:
Merge branch 'master' into branch-Level-7

* master: (34 commits)
  Touch up imports
  Revert "Remove one blank line"
  Update JavaDoc comments and help message
  Update some imports logic flow
  Add deadline command java doc
  Add JavaDoc comments for bye command and command interface
  Update README.md
  Add list command
  Add JavaDoc comments to three classes
  Modify build.gradle with the right names for main class and jar files
  Add tests related to commands
  Refactor the packages
  Test with gradle build and make slight adjustments
  Manage imports and create task package
  Put classes into packages
  Refactor command classes
  Test with data/xizi.txt
  Add parser class
  Add UI class
  Remove one blank line
  ...

Ma-Xizi committed:
Add slight adjustments to imports

Ma-Xizi committed:
Implement level-9

Ma-Xizi committed:
Merge branch 'master' into branch-A-JavaDoc

* master:
  Implement level-9

# Conflicts:
#	src/main/java/xizi/chatbot/Parser.java

Ma-Xizi committed:
Add more JavaDoc comments

Ma-Xizi committed:
Add more JavaDoc comments for task classes

xqtann commented on src/main/java/xizi/chatbot/command/CommandType.java:
```diff
@@ -0,0 +1,125 @@
+     * The command expects a task number as an argument.
+     * Example: "mark 1"
+     */
+    MARK("^mark (\\d+)$"),
```
I like how you made use of enum and regex to match user input with command type! 👍 😄

xqtann commented on src/main/java/xizi/chatbot/task/Todo.java:
```diff
@@ -0,0 +1,33 @@
+ * Represents a task that needs to be done without any specific deadline.
+ * This class extends the {@link Task} class and provides a simple task type.
+ */
+public class Todo extends Task{
```
Perhaps there should be a spacing after the class name `Task`?

xqtann commented on src/main/java/xizi/chatbot/Parser.java:
```diff
@@ -0,0 +1,96 @@
+            throw new XiziException("The task number does not exist. You have " + actions.getSize() + " tasks in total.");
+        }
+    }
+
```
Should this extra line be extracted out? 🤔

xqtann commented on src/main/java/xizi/chatbot/Storage.java:
```diff
@@ -0,0 +1,131 @@
+                tasks.add(task);
+            }
+        } finally {
+            reader.close(); // Ensure the reader is closed to prevent resource leaks
```
Great use of trailing comments to explain code! 👍

xqtann commented on src/main/java/xizi/chatbot/Storage.java:
```diff
@@ -0,0 +1,131 @@
+        String[] parts = line.split(" \\| ");
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy, h:mma"); // same as output format in xizi class
+        Task task;
+        switch (parts[0]) {
```
Great job on following coding standards for `switch case`!

xqtann reviewed (commented):
LGTM. Just some nits to fix. Overall code was easy to read and follow. Well done! 😄

Ma-Xizi committed:
Mimic JavaFX tutorial guides

LingZiyann commented on src/main/java/xizi/chatbot/Parser.java:
```diff
@@ -0,0 +1,96 @@
+ * It also interprets user commands and creates the corresponding {@code Command} instances.
+ */
+public class Parser {
+    private static final DateTimeFormatter INPUT_DATE_FORMAT = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```
Just a lil confused, the type is of DateTimeFormatter so it sounds like its a object that has methods to help to something but the varaible is called INPUT_DATE_FORMAT which sounds like a noun and does not have any methods to do anything. Idk if you get what i mean

LingZiyann commented on src/main/java/xizi/chatbot/Storage.java:
```diff
@@ -0,0 +1,131 @@
+     * @throws IOException If an I/O error occurs.
+     * @throws XiziException If the data in the file is corrupted or in an unexpected format.
+     */
+    public List<Task> load() throws IOException, XiziException {
```
method name could be more descriptive

Ma-Xizi replied:
Thanks! I have modified it to loadTasks() !

LingZiyann commented on src/main/java/xizi/chatbot/command/Command.java:
```diff
@@ -0,0 +1,28 @@
+     * @throws IOException   if an I/O error occurs during command execution
+     * @throws XiziException if a specific error related to the Xizi chatbot occurs
+     */
+    void execute(TaskList actions, Storage storage, Ui ui) throws IOException, XiziException;
```
method name could be more descriptive here

LingZiyann commented on src/main/java/xizi/chatbot/command/CommandType.java:
```diff
@@ -0,0 +1,125 @@
+ * Enum representing various types of commands that can be processed by the chatbot.
+ * Each command type is associated with a regular expression pattern used to match user input.
+ */
+public enum CommandType {
```
I like how u gave comments for every enum value

LingZiyann commented on src/main/java/xizi/chatbot/command/ListOnCommand.java:
```diff
@@ -0,0 +1,80 @@
+ * Represents a command to list all tasks that are scheduled on a specific date and time.
+ * This command filters tasks to show only those that are relevant to the given date.
+ */
+public class ListOnCommand implements Command {
```
Class name could be more descriptive.

LingZiyann reviewed (commented):
Good use of comments, naming standards, just a few names that can be named better

Ma-Xizi committed:
Implement JavaFx and migrate command handling to GUI

Ma-Xizi committed:
Apply basic styling as per the tutorial

Ma-Xizi committed:
Update coding style

Ma-Xizi committed:
Implement a rough idea of Varargs

Ma-Xizi committed:
Remove divider lines from bot responses for improved readability

After implementing GUI, String DIVIDER is no longer needed to
differentiate user input and bot response, as dialog boxes show
at different sides.

Refactor Ui class to remove dividers from bot responses. Let's
* Modify showLine() to display an empty line instead of "___".
* Remove divider lines from showWelcomeMessage() and other methods.
* Enhance readability of command output.

Ma-Xizi committed:
Add assertions to Xizi class

Ma-Xizi committed:
Improve code quality

Ma-Xizi committed:
Merge pull request #1 from Ma-Xizi/branch-A-Assertions

Add assertions to Xizi class

Ma-Xizi committed:
Add a getter in Task class

Ma-Xizi committed:
Implement "delete all completed task commands" function using streams

Ma-Xizi committed:
Merge pull request #2 from Ma-Xizi/branch-A-CodeQuality

Improve code quality

Ma-Xizi committed:
Merge pull request #3 from Ma-Xizi/branch-A-Streams

Implement "delete all completed task commands" function using streams

Ma-Xizi committed:
Remove duplicate methods

Ma-Xizi committed:
Attempt to set up Continuous Integration (CI) using .yml files

Ma-Xizi committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Attempt to set up Continuous Integration (CI) using .yml files

Ma-Xizi committed:
Remove outdated tests

Ma-Xizi committed:
Remove tests that print varied results on different systems

Ma-Xizi committed:
Remove unused imports

Ma-Xizi committed:
Implement a basic add tag function

Ma-Xizi committed:
Implement a simple 'remove tag' function

Ma-Xizi committed:
Implement 'search by tag' function and make '#' in tag commands optional

Ma-Xizi committed:
Recover data after repair

Ma-Xizi committed:
Experiment with README.doc styles

Ma-Xizi committed:
Debug about 'list on' logic to search for deadline tasks correctly

Ma-Xizi committed:
Add more tests and tweak some logics to catch more errors

Ma-Xizi committed:
Improve coding quality

Ma-Xizi committed:
Refractor to give better names

Ma-Xizi committed:
Improve code quality

Ma-Xizi committed:
Break up a long method to shorter methods

Ma-Xizi committed:
Improve code quality

Ma-Xizi committed:
Add more testing methods

Ma-Xizi committed:
Add tests for RemoveTagCommand class

Ma-Xizi committed:
Refractor the names of tests to match a consistent coding style

Ma-Xizi committed:
Rectify a small typo in the comment

Ma-Xizi committed:
Update README.md

Ma-Xizi committed:
Update README.md
